alpha-lab/gpt4-eyes/index.md

```diff
@@ -0,0 +1,69 @@
+:::
+
+## Combining Eye Tracking With AI for Assistive Applications
+
+Unlike popular Large *Language* Models (LLMs), which can only interpret and interact with text, Large *Multimodal* Models (LMMs) are also able to understand images. This means chatbots, like GPT, can understand, describe and interpret the content of real-world scenes, including identifying different objects and how people might interact with them.
+
+When combined with eye tracking, this holds potential for assistive technologies. For example, it could help individuals with low vision to identify the objects at which they are gazing, highlight potential hazards in their surroundings, or assist people with ALS who face significant communication challenges.
+
+Here, we show you how to build a simple, explorative, gaze-based assistive tool using Neon or Pupil Invisible and a new LMM, GPT-4V.
+
+## What Tools Enable This
+
+OpenAI recently unveiled a set of new APIs at their [DevDay event,](https://openai.com/blog/new-models-and-developer-products-announced-at-devday) introducing GPT-4V, an extension of their most advanced large language model capable of [understanding images](https://platform.openai.com/docs/guides/vision). 
+We got access to the preview and decided to see what is capable of when integrated with our eye trackers. We used Neon and Pupil Invisible because they remove the barrier of calibration, making them more suitable for real-life applications, longer recording sessions, and thus use alongside GPT-4V for experimenting with assistive applications.
+
+We used our [real-time API](https://docs.pupil-labs.com/neon/real-time-api/introduction/) as it enables streaming of scene camera and gaze positions over the network in real-time. This meant we could grab a frame at a selected moment and send it to GPT4-V for further processing, informing the wearer about what they were seeing or whatever we wanted to ask it about the scene.
+
+## Steps
+
+To try this demo, you'll need a Neon or Pupil Invisible device, a computer/laptop with internet access, and an OpenAI key that has access to the latest GPT-4V model. 
+
+::: tip
+[Don’t have access to `gpt-4-vision-preview model`?](https://help.openai.com/en/articles/7102672-how-can-i-access-gpt-4) You may have to spend 1$ on their APIs to get access to this model.
+:::
+
+Head to this [script](https://gist.github.com/mikelgg93/46a02823e1e271087c3eb6b2ab2cb99d) and follow the installation instructions.
+
+## Scene Understanding in Real-Time
+
+After running the example and pressing the `Space` key when you’re gazing at something, a snapshot of what you’re looking at will be sent to GPT-4V, and you will receive a vocal response. The nature of the AI assistant's response will depend on the selected model **("ASDF")**. This could be `A`, a description of what you're gazing at; `S`, any potential danger; `D`, a guess of your intentions; or `F`, a more detailed description of the environment. 
```

Comment:
```suggestion
Look at something in your environment and press the `Space` key. A snapshot with gaze overlay will be sent to GPT-4V, and you will receive a vocal response. 

The nature of the AI assistant's response will depend on the selected model. We have defined 4 modes that can be toggled with **("ASDF")** keys:
1. `A`: a description of what you're gazing at.
2. `S`: any potential danger. 
3. `D`: a guess about your intentions.
4. `F`: a more detailed description of the environment. 
```

Reply:
I think we could probably reduce the first paragraph a bit because we already talk about the architecture of the system a bit (e.g. how it works). 

Should we link to the prompts for each of the modes?

Reply:
It was decided to be left as an implementation detail in the gist